subgraphs/euler-finance/src/mappings/helpers.ts

```diff
@@ -794,6 +810,137 @@ export function processReward(epoch: _Epoch, epochStartBlock: BigInt, event: eth
+      const mkt = Market.load(mktID);
+      if (!mkt) {
+        log.error("[handleStake]market {} doesn't exist, but this should not happen at tx ={}", [
+          mktID,
+          event.transaction.hash.toHexString(),
+        ]);
+        continue;
+      }
+
+      // eIP28 blacklist FTT market from EUL distribution
+      // https://snapshot.org/#/eulerdao.eth/proposal/0x40874e40bc18ff33a9504a770d5aadfa4ea8241a64bf24a36777cb5acc3b59a7
+      if (epochID >= 16 && mkt.inputToken == FTT_ADDRESS) {
+        mkt._stakedAmount = BIGINT_ZERO;
+        mkt._weightedStakedAmount = BIGINT_ZERO;
+      }
+
+      const stakedAmount = mkt._stakedAmount;
+      let mktWeightedStakedAmount = BIGINT_ZERO;
+      if (isMarketEligible(mkt) && stakedAmount.gt(BIGINT_ZERO)) {
+        // finalized mkt._weightedStakedAmount for the epoch just ended
+        // epochStartBlock.minus(BIGINT_ONE) is the end block of prev epoch
+        updateWeightedStakedAmount(mkt, epochStartBlock.minus(BIGINT_ONE));
+        mktWeightedStakedAmount = mkt._weightedStakedAmount!;
+      }
+      mkt.save();
+
+      sumAccumulator = sumAccumulator.plus(mktWeightedStakedAmount.sqrt().toBigDecimal());
+    }
+
+    const EULPriceUSD = getEULPriceUSD(event);
+    const borrowerRewardToken = getOrCreateRewardToken(Address.fromString(EUL_ADDRESS), RewardTokenType.BORROW);
+    const lenderRewardToken = getOrCreateRewardToken(Address.fromString(EUL_ADDRESS), RewardTokenType.DEPOSIT);
+    const totalRewardAmount = BigDecimal.fromString((EUL_DIST[epochID - START_EPOCH] * EUL_DECIMALS).toString());
+
+    // scale to daily emission amount
+    const dailyScaler = BigDecimal.fromString((BLOCKS_PER_DAY / (BLOCKS_PER_EPOCH as f64)).toString());
+    const EUL_DECIMALS_BD = BigDecimal.fromString(EUL_DECIMALS.toString());
+    for (let i = 0; i < protocol._marketIDs!.length; i++) {
+      const mktID = protocol._marketIDs![i];
+      const mkt = Market.load(mktID);
+      if (!mkt) {
+        log.error("[handleStake]market {} doesn't exist, but this should not happen | tx ={}", [
+          mktID,
+          event.transaction.hash.toHexString(),
+        ]);
+        continue;
+      }
+      if (mkt.rewardTokens && mkt.rewardTokens!.length > 0) {
+        // reset reward emissions for the epoch
+        mkt.rewardTokenEmissionsAmount = [BIGINT_ZERO, BIGINT_ZERO];
+        mkt.rewardTokenEmissionsUSD = [BIGDECIMAL_ZERO, BIGDECIMAL_ZERO];
+      }
+
+      const rewardTokens: string[] = [];
+      const rewardTokenAmount = [BIGINT_ZERO, BIGINT_ZERO];
+      const rewardTokenUSD = [BIGDECIMAL_ZERO, BIGDECIMAL_ZERO];
+      // eIP24: 8000 EUL borrower rewards each for USDC, USDT, WETH, and WstETH
+      if ([USDC_ADDRESS, USDT_ADDRESS, WETH_ADDRESS, WSETH_ADDRESS].includes(mkt.inputToken)) {
+        rewardTokens.push(borrowerRewardToken.id);
+        rewardTokenAmount[0] = BigDecimalTruncateToBigInt(
+          BigDecimal.fromString("8000").times(EUL_DECIMALS_BD).times(dailyScaler),
```

Comment:
Isn't this apart of the reward distribution array?

Reply:
Oddly, it is not. Their [eIP24](https://snapshot.org/#/eulerdao.eth/proposal/0x7e65ffa930507d9116ebc83663000ade6ff93fc452f437a3e95d755ccc324f93) singles out these markets for special allocation of 8000 EUL.